src/pages/create-option.tsx

```diff
@@ -90,23 +78,33 @@ function CreateOption() {
 
         return (
             !missingFormFields &&
-            formState.premium !== 0 &&
+            parseFloat(formState.premium) !== 0 &&
             formState.premium != null &&
-            formState.strikePrice !== 0 &&
+            parseFloat(formState.strikePrice) !== 0 &&
             formState.strikePrice != null &&
-            formState.interval &&
-            !isBefore(formState.interval, new Date())
+            formState.interval
         );
     };
 
+    const handleChangeInterval = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setFormState((prev) => ({
+            ...prev,
+            interval: Math.max(1, Math.min(parseInt(event.target.value), 30)),
+        }));
+    };
+
     const handlePublishOption = () => {
+        const ethToWei = 1000000000000000000;
+        const strikePriceWei = parseFloat(formState.strikePrice) * ethToWei;
```

Comment:
you can do 
```ethers.utils.parseEther(`${parseFloat(formState.premium)} ether`)```

instead of `parseFloat(formState.premium)} * wei`

can remove wei altogether